nbodykit/io/csv.py

```diff
@@ -1,17 +1,136 @@
 import numpy
-
+import os
+from pandas import read_csv 
+    
 from ..extern.six import string_types
 from . import FileType, tools
 
+class CSVPartition(object):
```

Comment:
I'm using this class to cache the partitions through the ``value`` attribute, which calls read_csv on the string of bytes when requested